src/callback/callback_utils.cairo

```diff
@@ -63,73 +84,145 @@ fn set_saved_callback_contract(
 /// * `market` - The market to set callback contract for.
 fn get_saved_callback_contract(
     data_store: IDataStoreSafeDispatcher, account: ContractAddress, market: ContractAddress
-) -> ContractAddress { // TODO
-    0.try_into().unwrap()
+) -> Result<ContractAddress, Array<felt252>> {
+    data_store.get_address(keys::saved_callback_contract_key(account, market))
 }
 
 /// Called after a deposit execution.
 /// # Arguments
 /// * `key` - They key of the deposit.
 /// * `deposit` - The deposit that was executed.
 /// * `event_data` - The event log data.
-fn after_deposit_execution(key: felt252, deposit: Deposit, event_data: EventLogData) { // TODO
+fn after_deposit_execution(key: felt252, deposit: Deposit, event_data: EventLogData) {
+    if !is_valid_callback_contract(deposit.callback_contract) {
+        return;
+    }
+    let dispatcher = IDepositCallbackReceiverSafeDispatcher {
+        contract_address: deposit.callback_contract
+    };
+    match dispatcher.after_deposit_execution(key, deposit, event_data) {
+        Result::Ok => {},
+        Result::Err => emit_event(selector!("AfterDepositExecutionError"), key),
```

Comment:
e.g. here the event will never be emitted because the tx with revert if the callback fails

Reply:
sure but why not have the correct code from the start so it may work without modification once starknet is upgraded?

Reply:
Given that it's going to take a while I'm not sure this is the correct approach, especially given that it's not going to be caught in the tests

Reply:
you make a good point but since there are 482 uses of SafeDispatchers in the repo, i'm not sure this is the place to address this

Reply:
yeah I just jumped on the opportunity as I saw it here. cc @abdelhamidbakhta